### PR TITLE
Prepare v0.13 release

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Preparing the release of FirstUseAuthenticator v0.13.

Thanks to the amazing work done in oauthenticator and chartpress (:heart:), we can follow the same release pattern for FirstUseAuthenticator.

ToDos: 
- [x] release tagged builds to PyPi using Travis
- [x] add a changelog.md
- [x] add a release.md describing the release steps

Closes https://github.com/jupyterhub/firstuseauthenticator/issues/26

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,9 +4,20 @@ python:
...
+deploy:
+  provider: pypi
+  user: "__token__"
+  # password: set by TravisCI's environment variable PYPI_PASSWORD
+  distributions: sdist bdist_wheel
+  on:
+    # Only deploy on tagged commits instead of the default of only doing it to
+    # the master branch. A tag does not belong specifically to a branch, so
+    # without this it would fail to deploy for tags.
+    tags: true
```
[the remark this answers is not in the document]
Comment (GeorgianaElena) [reply .travis.yml]: Thanks a lot for clarifying this for me @consideRatio. I don't fully understand travis, so I didn't realised I was deploying multiple times. I read through kubespawner, oauthenticator and travis docs and it started to make more sense now. There's one thing though that's still not very clear for me. Is there any difference between *jobs* and *matrix* or any specific use-cases when one should use one or the other?

I added a new commit to address your feedback. Please let me know if you spot anything off or missing.

File: firstuseauthenticator/_version.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+version_info = (
+    0,
+    13,
+    0,
+    'dev',  # comment-out this line for a release
+)
+__version__ = '.'.join(map(str, version_info[:3]))
```
Comment (consideRatio) [inline firstuseauthenticator/_version.py]: Will this line work when commenting out dev, or show index out of bounds? Im not sure but i think we should exclude the index selector of [:3] here. But hmmm, i saw this pattern recently in jupyterhubs main repo, and then there when i made a 1.1.0b1 release and bumped it sgter to 1.1.0b2.dev.

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+   ```shell
+   git tag -a $VERSION -m $VERSION HEAD~1
+
+   # then verify you tagged the right commit
+   git log
+
+   # then push it
+   git push $ORIGIN refs/tags/$VERSION
+   ```
```
Comment (consideRatio) [inline RELEASE.md]: Nitpick: add blank line in the end

File: firstuseauthenticator/_version.py (hunk 2 above)
Comment (GeorgianaElena) [reply firstuseauthenticator/_version.py]: The slicing takes the first 3 elements from the tuple (it doesn't index the element ```version_info[3]```), so it won't go out of bounds ([docs ref](https://docs.python.org/3.5/library/stdtypes.html#sequence-types-list-tuple-range): Note no 4 )

It looked suspicious to me too when I first saw this in [oauthenticator](https://github.com/jupyterhub/oauthenticator/blob/master/oauthenticator/_version.py), but I checked it in the interpreter and later with the docs and it works.

**Comment by consideRatio** [comment]
Nice work! Time to try out the release instructions now then! :)

**Comment by GeorgianaElena** [comment]
@consideRatio, I addressed your feedback (thanks a lot for your patience :dog:).

I added the changelog also, but there is something I don't fully understand.
* Right now, there are "unreleased" changes in the changelog that will later be in v0.13. After this PR is merged, according to the release instructions, before making the release I should modify the changelog and add the changes as v0.13, right? How do I properly link the GitHub listing of changes in the changelog if the release 0.13 doesn't exist at the time I modify the file?
* Should https://github.com/jupyterhub/firstuseauthenticator/pull/21 be in this release? I will try to remember what I did in that PR (it's been a while since I last checked it out) and address your feedback. I'm ok if we wait for the next release also.

**Comment by consideRatio** [comment]
You can reference a tag like `[0.13.0]` and github will figure it out when viewing this markdown file on github itself.

- Make a commit without a PR to update the changelog with release date and make the [unreleased] section become a [0.13.0] section etc.
- Make a commit with the version changes etc
- Make a commit to reset the version to 0.13.1.dev (message "back to dev" has typically been used)

Push changes. Tag the pushed commit with version being 0.13.0. Push the tag reference. Verify travis builds and deploys a PyPI release successfully.

 before you make the version changes that you also commit and push, and then later tag to be 0.13.0.

---

Lets do more quick increments instead and let 0.14.0 follow quickly after 0.13.0 with that unmerged PR. I'd suggest you give it a self-review again, ping me who tries to review it without much insight but do my best anyhow, and the  we merge it if it seems ok to us.

**Comment by GeorgianaElena** [comment]
@consideRatio, shouldn't  travis trigger for both "release 0.13.0" and "back to dev" commits?
https://travis-ci.org/jupyterhub/firstuseauthenticator/builds?utm_medium=notification&utm_source=github_status

**Comment by consideRatio** [comment]
No if you push once with 100 commits it will only run once for the last commit. If you push 100 commits and two tags referencing two of these commits, it will run three times, one per tag and the final commit.

If a tag was or wasnt prefixed v is of very low importance i think, we dont read the tag name specifically at any place in our CI/CD pipeline currently. The PyPI released version is set in setup.py (referenced from someplace else)

**Comment by consideRatio** [comment]
:heart: :tada: excellent work!

**Comment by GeorgianaElena** [comment]
I started panicking when I saw that my release looked different from the other ones here https://github.com/jupyterhub/firstuseauthenticator/releases :smile_cat:

Thanks a lot for your help and patience @consideRatio :heart:

**Pull request merged**